### PR TITLE
Application gateway: disabled rule groups

### PR DESCRIPTION
**Pull request by timja**
Currently WIP

Having issues with the flatten, although the creation works fine

```
--- FAIL: TestAccAzureRMApplicationGateway_webApplicationFirewallDisabledRuleGroups (264.94s)
    testing.go:568: Step 0 error: errors during apply:

        Error: Error setting `waf_configuration`: waf_configuration.0.disabled_rule_group.0: '' expected a map, got 'slice'

          on /var/folders/6r/vwg96jz13pn1dlb6tk2_x6rm0000gn/T/tf-test386488536/main.tf line 48:
          (source code not available)


    testing.go:629: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: errors during refresh: Error setting `waf_configuration`: waf_configuration.0.disabled_rule_group.0: '' expected a map, got 'slice'

        State: <nil>
FAIL
```

Opening for comment on the design and any general feedback

```
  waf_configuration {
    enabled          = true
    firewall_mode    = "Detection"
    rule_set_type    = "OWASP"
    rule_set_version = "3.0"
    file_upload_limit_mb = 100
    request_body_check = true
    max_request_body_size_kb = 100

	disabled_rule_group {
		rule_group_name = "REQUEST-913-SCANNER-DETECTION"
		rules = [ 913100 ]
    }

	disabled_rule_group {
		rule_group_name = "REQUEST-920-PROTOCOL-ENFORCEMENT"
    }
  }
```

### CODE REVIEW

File: azurerm/resource_arm_application_gateway.go [hunk 1]
```diff
@@ -2657,11 +2703,42 @@ func flattenApplicationGatewayWafConfig(input *network.ApplicationGatewayWebAppl
...
+
+		output["rule_group_name"] = string(*v.RuleGroupName)
+
+		if v.Rules != nil {
+			for _, r := range *v.Rules {
+				rulesOutput := map[string]interface{}{}
+
+				rulesOutput["rules"] = r
+
+				result = append(result, rulesOutput)
```
Comment (mcharriere) [inline azurerm/resource_arm_application_gateway.go]: Shoudn't the rulesOutput be appended to the output instead of result?

Comment (timja) [reply azurerm/resource_arm_application_gateway.go]: Thanks for your help!

**Comment by bs-matil** [comment]
@timja would https://github.com/terraform-providers/terraform-provider-azurerm/pull/3394 do the trick for you?

**Comment by timja** [comment]
@bs-matil Yeah, there's one difference in that my PR allows optional rules so that the whole group is disabled.

I would like that supported as that matches the API docs

**Comment by timja** [comment]
Closing in favour of: https://github.com/terraform-providers/terraform-provider-azurerm/pull/3394

**Pull request closed without merging**